lib/Scheduler/aco.cpp

```diff
@@ -269,26 +263,23 @@ FUNC_RESULT ACOScheduler::FindSchedule(InstSchedule *schedule_out,
     pheremone_[i] = initialValue_;
   std::cerr << "initialValue_" << initialValue_ << std::endl;
 
-  InstSchedule *bestSchedule = InitialSchedule;
+  std::unique_ptr<InstSchedule> bestSchedule = std::move(InitialSchedule);
   if (bestSchedule) {
-    UpdatePheremone(bestSchedule);
+    UpdatePheremone(bestSchedule.get());
   }
   Config &schedIni = SchedulerOptions::getInstance();
   int noImprovementMax = schedIni.GetInt("ACO_STOP_ITERATIONS");
   int noImprovement = 0; // how many iterations with no improvement
   int iterations = 0;
   while (true) {
-    InstSchedule *iterationBest = NULL;
+    std::unique_ptr<InstSchedule> iterationBest;
     for (int i = 0; i < ants_per_iteration; i++) {
-      InstSchedule *schedule = FindOneSchedule();
+      std::unique_ptr<InstSchedule> schedule = FindOneSchedule();
       if (print_aco_trace)
-        PrintSchedule(schedule);
-      if (iterationBest == NULL ||
+        PrintSchedule(schedule.get());
+      if (iterationBest == nullptr ||
```

Comment:
How do we feel about just `!iterationBest`? That's what the CppCoreGuidelines says to do: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es87-dont-add-redundant--or--to-conditions

Reply:
The LLVM Coding standards don't say anything either way, and either way you have to add another operator `!` vs `== nullptr`.  ES87 only seems to cover causes where you are checking if it is != null, not causes where you are checking if it is null.